src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_TryCatch.cs

```diff
@@ -3342,6 +3342,63 @@ void F()
             VerifyFlowGraphForTest<BlockSyntax>(compilation, expectedGraph);
         }
 
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void TryFlow_29()
+        {
+            var source = @"
+#pragma warning disable CS0168
+#pragma warning disable CS0219
+class C
+{
+    void F()
+    /*<bind>*/{
+        try
+        {
+            int i = 1;
+        }
+        finally
+        {
+            int j;
+            goto label1;
+label1:     ;
+        }
+    }/*</bind>*/
+}";
+
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.RegularWithFlowAnalysisFeature);
+
+            compilation.VerifyDiagnostics();
+
+            string expectedGraph = @"
+Block[B0] - Entry
+    Statements (0)
+    Next (Regular) Block[B1]
+        Entering: {R1}
+
+.locals {R1}
+{
+    Locals: [System.Int32 i]
+    Block[B1] - Block
+        Predecessors: [B0]
+        Statements (1)
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32, IsImplicit) (Syntax: 'i = 1')
+              Left: 
+                ILocalReferenceOperation: i (IsDeclaration: True) (OperationKind.LocalReference, Type: System.Int32, IsImplicit) (Syntax: 'i = 1')
+              Right: 
+                ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 1) (Syntax: '1')
+
+        Next (Regular) Block[B2]
+            Leaving: {R1}
+}
+
+Block[B2] - Exit
```

Comment:
>Block[B2] [](start = 0, length = 9)

Should there be a block for `finally` and `j`?

Reply:
> Should there be a block for finally and j?

We intentionally remove try/finally when finally is a NoOp.

---
In reply to: [173525767](https://github.com/dotnet/roslyn/pull/25338#discussion_r173525767) [](ancestors = 173525767)